target_chains/ethereum/contracts/contracts/pyth/PythPriceOracleGetter.sol

```diff
@@ -0,0 +1,40 @@
+// SPDX-License-Identifier: Apache 2
+pragma solidity ^0.8.0;
+
+import "@pythnetwork/pyth-sdk-solidity/PythErrors.sol";
+
+import "../aave/interfaces/IPriceOracleGetter.sol";
+import "./PythAssetRegistry.sol";
+
+contract PythPriceOracleGetter is PythAssetRegistry, IPriceOracleGetter {
+    address public immutable override BASE_CURRENCY;
+    uint256 public immutable override BASE_CURRENCY_UNIT;
+
+    constructor(
+        address pyth,
+        address[] memory assets,
+        bytes32[] memory priceIds,
+        address baseCurrency,
+        uint256 baseCurrencyUnit
+    ) {
+        setPyth(pyth);
+        setAssetsSources(assets, priceIds);
+        setBaseCurrency(baseCurrency, baseCurrencyUnit);
+        BASE_CURRENCY = _registryState.BASE_CURRENCY;
+        BASE_CURRENCY_UNIT = _registryState.BASE_CURRENCY_UNIT;
+    }
+
+    /// @inheritdoc IPriceOracleGetter
+    function getAssetPrice(
+        address asset
+    ) external view override returns (uint256) {
+        bytes32 priceId = _registryState.assetsPriceIds[asset];
+        if (asset == BASE_CURRENCY) {
+            return BASE_CURRENCY_UNIT;
+        }
+        if (priceId == 0) {
+            revert PythErrors.PriceFeedNotFound();
+        }
+        return uint256(uint64(pyth().getPriceUnsafe(priceId).price));
```

Comment:
I think we likely need to do some price conversion here using the exponent (?). I'm not sure what exactly Aave expects the uint256 to be, but we need to find out how they're representing decimals and convert to that.

Reply:
and let's do a staleness check here (with a configurable threshold)

Reply:
it doesn't look like `AaveOracle` explicitly checks anything related to the exponent/decimals they just always assume that it matches and the values will be normalized correctly. I'll add an explicit conversion on ours to be safe.

also added a `validTimePeriodSeconds` in the constructor and use it in `getPriceNoOlderThan()` for `getAssetPrice`